jax/_src/ad_checkpoint.py

```diff
@@ -491,10 +491,11 @@ def f_(*args):
   jaxpr = jaxpr_.jaxpr
   out_shape = out_shape_[1]
   num_res = tree_structure(out_shape).num_leaves
-  jaxpr = jaxpr.replace(outvars=jaxpr.outvars[len(jaxpr.outvars) - num_res:])
-  out_tree = lambda: tree_structure(out_shape)
+  jaxpr = jaxpr.replace(
+      outvars=jaxpr.outvars[len(jaxpr.outvars) - num_res:],
+      debug_info=debug_info._replace(result_paths=None))
   assert len(jaxpr.invars) == len(in_leaves)
-  return _saved_residuals(jaxpr, debug_info.arg_names)
+  return _saved_residuals(jaxpr, debug_info.arg_names or ("",) * len(jaxpr.invars))
```

Comment:
Why not set it to `None` or `()`, rather than a tuple of empty strings?

Reply:
Maybe `(None,) * len(jaxpr.invars)`? That way the callee can annotate `arg_names: Sequence[str]` and the existing code handles it.

Reply:
I think that you meant `("None",) * len(jaxpr.invars)` (i.e., a tuple of `str`). How about `("unknown",) * len(jaxpr.invars)`?